src/checkout.c

```diff
@@ -1966,34 +1966,109 @@ static int conflict_entry_name(
 	return 0;
 }
 
+static int checkout_path_suffixed_0(
```

Comment:
The name is a bit awkward. What exactly does this function do that `checkout_path_suffixed` doesn't?